gr-blocks/lib/CMakeLists.txt

```diff
@@ -199,6 +199,8 @@ list(APPEND gr_blocks_sources
     wavfile_source_impl.cc
 )
 
+set_source_files_properties(file_source_impl.cc PROPERTIES COMPILE_FLAGS -D_FILE_OFFSET_BITS=64)
```

Comment:
I'm wondering if a `#define` in the implementation would be better?

Reply:
No. It doesn't work on 32-bit architectures.

Reply:
It works, but you have to put it before the #includes.

Reply:
Maybe that's better since it will keep the `#define` and the corresponding code together instead of pulling it out into the build system. 
As long as it works I'm personally fine with both.